src/OrleansCodeGenerator/CodeGeneratorCommon.cs

```diff
@@ -155,14 +155,27 @@ internal static string GenerateSourceCode(CompilationUnitSyntax code)
         /// <returns>Types which have corresponding generated classes marked with any of the provided marker attributes.</returns>
         internal static HashSet<Type> GetTypesWithImplementations(params Type[] markerAttributes)
         {
+            if (markerAttributes == null)
+            {
+                throw new ArgumentNullException("markerAttributes");
+            }
+
+            if (!markerAttributes.All(_ => typeof (GeneratedAttribute).IsAssignableFrom(_)))
+            {
+                throw new ArgumentException("Every element must be a subtype of GeneratedAttribute", "markerAttributes");
+            }
+
             // Get assemblies which contain generated code.
             var all =
                 AppDomain.CurrentDomain.GetAssemblies()
                     .Where(_ => _.GetCustomAttribute<GeneratedCodeAttribute>() != null)
                     .SelectMany(_ => _.DefinedTypes);
-
+            
             // Get all generated types in each assembly.
-            var attributes = all.SelectMany(_ => _.GetCustomAttributes()).OfType<GeneratedAttribute>();
+            var attributes =
+                all.SelectMany(_ => _.GetCustomAttributes())
+                    .Where(_ => markerAttributes.Contains(_.GetType()))
+                    .OfType<GeneratedAttribute>();
```

Comment:
Only if the code generator does not generate code for it, which it ought to if the attribute derives from `GeneratedAttribute`. I think removing it is nice and clean - it makes the code more approachable.